stable/k8s-spot-termination-handler/README.md

```diff
@@ -0,0 +1,57 @@
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` daemon set:
+
+```console
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Access control
+
+It is critical for the Kubernetes cluster to correctly setup access control of Kubernetes Dashboard. See this [guide](https://github.com/kubernetes/dashboard/wiki/Access-control) for best practises.
+
+It is highly recommended to use RBAC with minimal privileges needed for Spot Termination handler to run.
+
+## Configuration
+
+The following table lists the configurable parameters of the kubernetes-dashboard chart and their default values.
+
+| Parameter               | Description                                                                                                                 | Default                                       |
+|-------------------------|-----------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------|
+| `image.repository`      | Repository for container image                                                                                              | `quay.io/pusher/k8s-spot-termination-handler` |
+| `image.tag`             | Image tag                                                                                                                   | `v0.1.0`                                      |
+| `image.pullPolicy`      | Image pull policy                                                                                                           | `IfNotPresent`                                |
+| `nodeSelector`          | node labels for pod assignment                                                                                              | `{}`                                          |
+| `tolerations`           | List of node taints to tolerate (requires Kubernetes >= 1.6)                                                                | `[]`                                          |
+| `resources`             | Pod resource requests & limits                                                                                              | `{}`                                          |
+| `rbac.create`           | Create & use RBAC resources                                                                                                 | `true`                                        |
+| `serviceAccount.create` | Whether a new service account name that the agent will use should be created.                                               | `true`                                        |
+| `serviceAccount.name`   | Service account to be used. If not set and serviceAccount.create is `true` a name is generated using the fullname template. |                                               |
```

Comment:
I think the `nodeSelector` value is quite important here? From values.yaml:

```
nodeSelector:
  "node-role.kubernetes.io/spot-worker": "true"
```

Reply:
A question about this though: is this label a convention? How is does the label get there normally?

Reply:
As mentioned above

> removed node-role.kubernetes.io/spot-worker as default from nodeSelector, because a chart consumer can't easily remove that key and node-role.kubernetes.io/spot-worker is only a recommendation by the K8s Spot Termination handler. But the real node labels are dependent by the cluster administrator or maybe the chart consumer uses affinity.

So the default value is indeed `{}`, but in the `values.yaml` from my PR it's commented as "default" value. As said it depends on the administrator which label is used.

Any idea how to document a recommended "default" value in this table?

Reply:
I don't know how `node-role.kubernetes.io/spot-worker` is a convention. The recommendation in the readme of the spot termination handler follows a good convention with this key name, but it is not a requirement for the cluster administrator.

Reply:
Ah you are right, sorry. I didn't see it being removed.

I'm still curious if there is a convention in what label to use for this though. Is there a common choice? I know it's up to the admin but still.

Reply:
Quick search, cause I can't remember everything ^^ `node-role.kubernetes.io/<role-name>` is already a convention to define node roles, which are also shown in `kubectl get nodes`, but the role name is free to be defined by the cluster administrator.

Another reason, why it shouldn't be defaulted by the chart is helm/helm#3117, keys can't be removed atm from maps in chart values.